src/main/java/uk/gov/pay/products/persistence/dao/ProductDao.java

```diff
@@ -116,4 +116,32 @@ private List<ProductUsageStat> findProductsAndUsageQuery(Integer gatewayAccountI
         }
         return queryBuilder.getResultList();
     }
+
+    public List<ProductUsageStat> findUnusedProducts(Integer gatewayAccountId) {
+        return findUnusedProductsQuery(gatewayAccountId);
+    }
+
+    public List<ProductUsageStat> findUnusedProductsQuery(Integer gatewayAccountId) {
+        boolean shouldFilterGatewayAccount = gatewayAccountId != null;
+
+        String conditionalFilter = shouldFilterGatewayAccount ?
+                "AND product.gateway_account_id = ?1 " :
+                "";
+
+        String query = "SELECT product.* FROM products product " +
+                "LEFT JOIN payments payment ON product.id = payment.product_id " +
+                "WHERE payment.product_id IS NULL " +
+                conditionalFilter;
```

Comment:
could be a text block (untested)
```suggestion
 String query = """
        SELECT product.*
        FROM products product
        LEFT JOIN payments payment ON product.id = payment.product_id
        WHERE payment.product_id IS NULL
   """ + (gatewayAccountId != null ? "AND product.gateway_account_id = :gatewayAccountId" : "");
```